libdnf/dnf-context.cpp

```diff
@@ -2440,38 +2439,38 @@ gboolean
 dnf_context_remove(DnfContext *context, const gchar *name, GError **error) try
 {
     DnfContextPrivate *priv = GET_PRIVATE(context);
-    GPtrArray *pkglist;
-    hy_autoquery HyQuery query = NULL;
-    gboolean ret = TRUE;
-    guint i;
 
     /* create sack and add repos */
     if (priv->sack == NULL) {
         dnf_state_reset(priv->state);
-        ret = dnf_context_setup_sack(context, priv->state, error);
-        if (!ret)
+        if (!dnf_context_setup_sack(context, priv->state, error))
             return FALSE;
     }
 
-    /* find installed packages to remove */
-    query = hy_query_create(priv->sack);
-    query->installed();
-    hy_query_filter(query, HY_PKG_NAME, HY_EQ, name);
-    pkglist = hy_query_run(query);
+    g_auto(HySubject) subject = hy_subject_create(name);
+    g_auto(HySelector) selector = hy_subject_get_best_selector(subject, priv->sack, NULL, FALSE, NULL);
```

Comment:
What about to use `hy_subject_get_best_solution`. Be honest the approach using hy_selector_matches is not nice and too long. Also the code depend on assumption that there are not available packages.

Reply:
I recommend to even raise two types of error - no match and not installed because it is quite easy to get such an information.

Reply:
May be - `hy_subject_get_best_query` will be a better option, because it will not require to handle nevra object.

Reply:
Or even better would be to use -
```
libdnf::Query query(sack, libdnf::Query::ExcludeFlags::APPLY_EXCLUDES);
auto ret = query->filterSubject(subject, forms, icase, with_nevra, with_provides, with_filenames);
```

Reply:
I made a change. `query->filterSubject` is now used.